x/faucet/module.go

```diff
@@ -0,0 +1,118 @@
+package faucet
+
+import (
+	"encoding/json"
+
+	"github.com/gorilla/mux"
+	"github.com/spf13/cobra"
+
+	"github.com/cosmos/cosmos-sdk/client/context"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/public-awesome/stakebird/x/faucet/client/cli"
+	"github.com/public-awesome/stakebird/x/faucet/client/rest"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+// type check to ensure the interface is properly implemented
+var (
+	_ module.AppModule      = AppModule{}
+	_ module.AppModuleBasic = AppModuleBasic{}
+)
+
+// app module Basics object
+type AppModuleBasic struct {
+	// cdc codec.Marshaler
+}
+
+func (AppModuleBasic) Name() string {
+	return ModuleName
+}
+
+func (AppModuleBasic) RegisterCodec(cdc *codec.Codec) {
+	RegisterCodec(cdc)
+}
+
+func (AppModuleBasic) DefaultGenesis(_ codec.JSONMarshaler) json.RawMessage {
+	return nil
+}
+
+// Validation check of the Genesis
+func (AppModuleBasic) ValidateGenesis(_ codec.JSONMarshaler, bz json.RawMessage) error {
+	return nil
+}
+
+// Register rest routes
+func (AppModuleBasic) RegisterRESTRoutes(ctx context.CLIContext, rtr *mux.Router) {
+	if profile == TESTNET {
+		rest.RegisterRoutes(ctx, rtr, StoreKey)
+	}
+}
+
+// Get the root query command of this module
+func (AppModuleBasic) GetQueryCmd(cdc *codec.Codec) *cobra.Command {
+	return nil
+}
+
+// Get the root tx command of this module
+func (AppModuleBasic) GetTxCmd(cdc *codec.Codec) *cobra.Command {
+	if profile == TESTNET {
+		return cli.GetTxCmd(StoreKey, cdc)
+	}
+
+	return nil
+}
+
+type AppModule struct {
+	AppModuleBasic
+	keeper Keeper
+}
+
+// NewAppModule creates a new AppModule Object
+func NewAppModule(k Keeper) AppModule {
+	return AppModule{
```

Comment:
Do we want to return `nil` here if `profile == TESTNET`? Like we do with the handler?

Reply:
Should be fine because although the module will be wired up, there'll be no way to access it since it'll be disabled by the handler and other entrypoints.

Reply:
That's what I thought... though part of me wonders what would happen if we add something like CosmWasm in the future. I'm not too worried though, we can cross that bridge when we get there.